### ISSUE TITLE
get-discovery-doc generates incorrect discovery document

### ISSUE THREAD

**Issue by sparhomenko** [cloudendpoints/endpoints-java#31]
When using the latest (`2.0.0-beta.10`) version of the endpoints tool to generate a Java client library we started getting compilation problems in the code that uses this library. Comparing the `get-discovery-doc` result with the previous version we found the following differences:
1. When setting `namespace` attribute of the `@Api` annotation:
    ```java
    @Api(name = "<api>", namespace = @ApiNamespace(ownerDomain = "<company>.com", ownerName = "<Company>", packagePath = "<project>"))
    ```
    the old version reflects it in the discovery document:
    ```json
     "ownerDomain": "<company>.com",
     "ownerName": "<Company>",
     "packagePath": "<project>",
    ```
    while new version does not add these keys into the discovery document at all, causing the generated Java code to use `com.appspot.myapi.<api>` package instead of `com.<company>.<project>.<api>`.
2. Old version takes `<application>` from `WEB-INF/appengine-web.xml` into account:
    ```json
    "baseUrl": "https://<application>.appspot.com/_ah/api/<api>/<version>/<resource>/",
    "rootUrl": "https://<application>.appspot.com/_ah/api/",
    ```
    and the new version does not:
    ```json
    "baseUrl": "https://myapi.appspot.com/_ah/api/<api>/<version>/<resource>/",
    "rootUrl": "https://myapi.appspot.com/_ah/api/",
    ```
3. For an API method with two parameters:
    ```java
    @ApiMethod(path = "parent/{parent}/child/{child}")
    public void updateParent(@Named("parent") String parent, @Named("child") int child) {
        ...
    }
    ```
   the old version correctly reflects the parameter order:
   ```json
    "parameterOrder": [
      "parent",
      "child"
     ],
    ```
    while the new version reverses the order, probably ordering alphabetically:
   ```json
    "parameterOrder": [
      "child",
      "parent"
     ],
    ```

Looking at the recent commits it seems this is caused by the new `LocalDiscoveryProvider` introduced in `2.0.0-beta.10`. It's not a blocker for us as we can always downgrade, but the new functionality would actually be useful: our normal Maven build regenerates the discovery document and Java client library, and doing that locally would help to improve the build reliability.

**Comment by tangiel** [comment]
Thanks for the report. Please downgrade for now, and we'll fix it in the next release.

**Comment by tangiel** [comment]
These should all be fixed by #34 and is being released in beta.12. Thank you for the report. Please let us know if there are still issues.

**Comment by sparhomenko** [comment]
The fix looks good, thanks for your help.
